src/python/pants/backend/python/macros/poetry_requirements.py

```diff
@@ -330,10 +330,16 @@ def parse_pyproject_toml(pyproject_toml: PyProjectToml) -> set[Requirement]:
     # See: https://python-poetry.org/docs/pyproject/#dependencies-and-dev-dependencies
     dependencies.pop("python", None)
 
+    groups = poetry_vals.get("group", {})
+    group_deps = {}
+
+    for group in groups.values():
+        group_deps.update(**group.get("dependencies", {}))
+
     dev_dependencies = poetry_vals.get("dev-dependencies", {})
-    if not dependencies and not dev_dependencies:
+    if not dependencies and not dev_dependencies and not group_deps:
         logger.warning(
-            "No requirements defined in poetry.tools.dependencies and "
+            "No requirements defined in any Poetry dependency groups, poetry.tools.dependencies and "
             f"poetry.tools.dev-dependencies in {pyproject_toml.toml_relpath}, which is loaded "
```

Comment:
Should these be `tool.poetry.dependencies`, rather than `poetry.tools`? Ditto dev deps

Reply:
I actually only added the "any Poetry dependency groups" line; poetry.tools was what was already there so I think that was a typo from the original PR.